libsolidity/analysis/TypeChecker.cpp

```diff
@@ -744,14 +744,25 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 	)
 		m_errorReporter.typeError(_variable.location(), "Internal or recursive type is not allowed for public state variables.");
 
-	if (varType->category() == Type::Category::Array)
+	switch (varType->category())
+	{
+	case Type::Category::Array:
 		if (auto arrayType = dynamic_cast<ArrayType const*>(varType.get()))
 			if (
 				((arrayType->location() == DataLocation::Memory) ||
 				(arrayType->location() == DataLocation::CallData)) &&
 				!arrayType->validForCalldata()
 			)
 				m_errorReporter.typeError(_variable.location(), "Array is too large to be encoded.");
+		break;
+	case Type::Category::Mapping:
+		if (auto mappingType = dynamic_cast<MappingType const*>(varType.get()))
+			if (mappingType->keyType()->isDynamicallySized())
+				m_errorReporter.typeError(_variable.location(), "Dynamically-sized mapping keys not yet implemented.");
```

Comment:
I think we shouldn't make any commitment here, just state `Dynamically-sized mapping keys are not supported.`